internal/provider/providers/aws/provider.go

```diff
@@ -0,0 +1,184 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/netip"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/route53"
+	"github.com/aws/aws-sdk-go-v2/service/route53/types"
+
+	"github.com/qdm12/ddns-updater/internal/models"
+	"github.com/qdm12/ddns-updater/internal/provider/constants"
+	"github.com/qdm12/ddns-updater/internal/provider/errors"
+	"github.com/qdm12/ddns-updater/internal/provider/utils"
+	"github.com/qdm12/ddns-updater/pkg/publicip/ipversion"
+)
+
+type Provider struct {
+	awsAccessKeyID     string
+	awsSecretAccessKey string
+	domain             string
+	host               string
+	hostedZoneID       string
+	ipVersion          ipversion.IPVersion
+	ipv6Suffix         netip.Prefix
+	region             string
+	ttl                int64
+}
+
+func New(data json.RawMessage, domain, host string,
+	ipVersion ipversion.IPVersion, ipv6Suffix netip.Prefix) (
+	provider *Provider, err error) {
+	var providerSpecificSettings settings
+	err = json.Unmarshal(data, &providerSpecificSettings)
+	if err != nil {
+		return nil, fmt.Errorf("decoding provider specific settings: %w", err)
+	}
+
+	err = validateSettings(providerSpecificSettings, domain, host)
+	if err != nil {
+		return nil, fmt.Errorf("validating provider specific settings: %w", err)
+	}
+	ttl := int64(300)
+	if providerSpecificSettings.TTL != nil {
+		ttl = *providerSpecificSettings.TTL
+	}
+
+	region := "us-east-1"
+	if providerSpecificSettings.Region != "" {
+		region = providerSpecificSettings.Region
+	}
```

Comment:
should we enforce the region to be set perhaps? What's the reason to default to `us-east-1`?

Reply:
No specific reason for that. I just got watherver region UI was showing at the moment for the 'Global Access', I will make it mandatory.

Reply:
I found out that signing requests for Global Resources must be `us-east-1`. As the goal is to `UPSERT`, the ddns-updater should make requests only to `us-east-1`. I will remove the Region from options.

Reply:
Updated the region to be `us-east-1` due to the limitations. Please, take another look